diff --git a/node_modules/@basalt/twig-renderer/dist/TwigRenderer.php b/node_modules/@basalt/twig-renderer/dist/TwigRenderer.php
index 43eefae..474fd6e 100644
--- a/node_modules/@basalt/twig-renderer/dist/TwigRenderer.php
+++ b/node_modules/@basalt/twig-renderer/dist/TwigRenderer.php
@@ -48,9 +48,10 @@ class TwigRenderer {
 
   private function createTwigEnv($loaders) {
     $twig = new \Twig_Environment($loaders, [
-      'debug' => $this->config['debug'],
+      'debug' => false,
       'autoescape' => $this->config['autoescape'],
-      'cache' => false, // @todo Implement Twig caching
+      'auto_reload' => true,
+      'cache' => 'cache/twig-renderer', // @todo Implement Twig caching
     ]);
 
     if (isset($this->config['alterTwigEnv'])) {
