diff --git a/node_modules/scss-comment-parser/index.js b/node_modules/scss-comment-parser/index.js
index 62d3dfa..c031ff2 100644
--- a/node_modules/scss-comment-parser/index.js
+++ b/node_modules/scss-comment-parser/index.js
@@ -188,7 +188,7 @@ var filterAndGroup = function (lines) {
   return nLines;
 };
 
-var extractor = new CDocParser.CommentExtractor(scssContextParser, { blockComment : false});
+var extractor = new CDocParser.CommentExtractor(scssContextParser);
 
 var Parser = function (annotations, config) {
   this.commentParser = new CDocParser.CommentParser(annotations, config);
