diff --git a/src/PatternLab/PatternEngine/Twig/Loaders/StringLoader.php b/src/PatternLab/PatternEngine/Twig/Loaders/StringLoader.php
index 4526c2ce7..87a76068e 100644
--- a/src/PatternLab/PatternEngine/Twig/Loaders/StringLoader.php
+++ b/src/PatternLab/PatternEngine/Twig/Loaders/StringLoader.php
@@ -26,6 +26,7 @@ class StringLoader extends Loader {
 		
 		// set-up the defaults
 		$twigDebug = Config::getOption("twigDebug");
+		$twigCache = Config::getOption("twigCache");
 		
 		// go through various places where things can exist
 		$filesystemLoaderPaths = array();
@@ -52,7 +53,11 @@ class StringLoader extends Loader {
 		
 		// set-up Twig
 		$twigLoader = new \Twig_Loader_Chain($loaders);
-		$instance   = new \Twig_Environment($twigLoader, array("debug" => $twigDebug));
+		$instance   = new \Twig_Environment($twigLoader, array(
+			"debug" => false,
+			"cache" => $twigCache,
+			"auto_reload" => true,
+		));
 		
 		// customize Twig
 		TwigUtil::setInstance($instance);
@@ -91,3 +96,4 @@ class StringLoader extends Loader {
 	}
 	
 }
+
