diff --git a/node_modules/@bolt/build-utils/LICENSE b/node_modules/@bolt/build-utils/LICENSE
deleted file mode 100644
index 0946736..0000000
--- a/node_modules/@bolt/build-utils/LICENSE
+++ /dev/null
@@ -1,21 +0,0 @@
-MIT License
-
-Copyright (c) 2019 Pegasystems
-
-Permission is hereby granted, free of charge, to any person obtaining a copy
-of this software and associated documentation files (the "Software"), to deal
-in the Software without restriction, including without limitation the rights
-to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-copies of the Software, and to permit persons to whom the Software is
-furnished to do so, subject to the following conditions:
-
-The above copyright notice and this permission notice shall be included in all
-copies or substantial portions of the Software.
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-SOFTWARE.
diff --git a/node_modules/@bolt/build-utils/manifest.js b/node_modules/@bolt/build-utils/manifest.js
index d1e797f..a1f7023 100644
--- a/node_modules/@bolt/build-utils/manifest.js
+++ b/node_modules/@bolt/build-utils/manifest.js
@@ -51,7 +51,7 @@ try {
   ).version;
 } catch (error) {
   boltManifest.version = JSON.parse(
-    fs.readFileSync(path.join(__dirname, '../package.json'), 'utf8'),
+    fs.readFileSync(path.join(__dirname, 'package.json'), 'utf8'),
   ).version;
 }
 
