diff --git a/node_modules/@bolt/build-tools/create-webpack-config.js b/node_modules/@bolt/build-tools/create-webpack-config.js
index f0cf432..3c456b3 100644
--- a/node_modules/@bolt/build-tools/create-webpack-config.js
+++ b/node_modules/@bolt/build-tools/create-webpack-config.js
@@ -150,7 +150,7 @@ async function createWebpackConfig(buildConfig) {
     const globalEntryName = 'bolt-global';
 
     if (components.global) {
-      entry[globalEntryName] = [];
+      entry[globalEntryName] = ['@bolt/core/styles/index.scss'];
 
       components.global.forEach(component => {
         if (component.assets.style) {
