diff --git a/node_modules/@bolt/core/polyfills/index.js b/node_modules/@bolt/core/polyfills/index.js
index 6b618a7..ea7a5e1 100644
--- a/node_modules/@bolt/core/polyfills/index.js
+++ b/node_modules/@bolt/core/polyfills/index.js
@@ -1,15 +1,15 @@
 import 'es6-promise/auto';
 import 'element-closest';
 import 'mdn-polyfills/Node.prototype.prepend';
-import 'core-js/modules/es6.array.iterator';
-import 'core-js/modules/es6.array.from';
-import 'core-js/modules/es6.string.starts-with';
-import 'core-js/modules/es7.array.includes';
-import 'core-js/modules/es6.array.for-each';
-import 'core-js/modules/es6.object.assign';
-import 'core-js/modules/es6.string.includes';
+import 'core-js/modules/es.array.iterator';
+import 'core-js/modules/es.array.from';
+import 'core-js/modules/es.string.starts-with';
+import 'core-js/modules/es.array.includes';
+import 'core-js/modules/es.array.for-each';
+import 'core-js/modules/es.object.assign';
+import 'core-js/modules/es.string.includes';
 import './custom-event-polyfill';
-import 'core-js/modules/es6.array.find';
+import 'core-js/modules/es.array.find';
 import './symbol-polyfill';
 import './remove-polyfill';
 import '@webcomponents/template/template.js';
