diff --git a/node_modules/lazysizes/lazysizes.js b/node_modules/lazysizes/lazysizes.js
index 26ba477..076c71e 100644
--- a/node_modules/lazysizes/lazysizes.js
+++ b/node_modules/lazysizes/lazysizes.js
@@ -97,7 +97,12 @@
 	};
 
 	var getCSS = function (elem, style){
-		return (getComputedStyle(elem, null) || {})[style];
+    try {
+      return (getComputedStyle(elem, null) || {})[style];
+    }
+    catch {
+      return {}[style];
+    }
 	};
 
 	var getWidth = function(elem, parent, width){
@@ -586,8 +591,21 @@
 			_: function(){
 				started = Date.now();
 
-				lazysizes.elements = document.getElementsByClassName(lazySizesConfig.lazyClass);
-				preloadElems = document.getElementsByClassName(lazySizesConfig.lazyClass + ' ' + lazySizesConfig.preloadClass);
+        lazysizes.elements = lazySizesConfig.getElements
+          ? lazySizesConfig.getElements(`.${lazySizesConfig.lazyClass}`)
+          : document.getElementsByClassName(lazySizesConfig.lazyClass);
+
+        preloadElems = lazySizesConfig.getElements
+          ? lazySizesConfig.getElements(
+              `.${lazySizesConfig.lazyClass}.${
+                lazySizesConfig.preloadClass
+              }`,
+            )
+          : document.getElementsByClassName(
+              lazySizesConfig.lazyClass +
+                ' ' +
+                lazySizesConfig.preloadClass,
+            );
 
 				addEventListener('scroll', throttledCheckElements, true);
 
