diff --git a/node_modules/get-own-property-symbols/build/get-own-property-symbols.max.js b/node_modules/get-own-property-symbols/build/get-own-property-symbols.max.js
index 02ddd12..2a3c1d1 100644
--- a/node_modules/get-own-property-symbols/build/get-own-property-symbols.max.js
+++ b/node_modules/get-own-property-symbols/build/get-own-property-symbols.max.js
@@ -153,7 +153,21 @@ THE SOFTWARE.
   descriptor.value = $getOwnPropertySymbols;
   defineProperty(Object, GOPS, descriptor);
 
+  /**
+   * @see https://github.com/aurelia/polyfills/pull/52/files
+   */
+  var cachedWindowNames = typeof window === 'object' ? Object.getOwnPropertyNames(window) : [];
+  var originalObjectGetOwnPropertyNames = Object.getOwnPropertyNames;
   descriptor.value = function getOwnPropertyNames(o) {
+    if (toString.call(o) === '[object Window]') {
+      try {
+        return originalObjectGetOwnPropertyNames(o);
+      } catch (e) {
+        // IE bug where layout engine calls userland gOPN for cross-domain "window" objects
+        return [].concat([], cachedWindowNames);
+      }
+    }
+
     return gOPN(o).filter(onlyNonSymbols);
   };
   defineProperty(Object, GOPN, descriptor);
